mmv1/third_party/terraform/acctest/tgc_utils.go

```diff
@@ -62,3 +63,91 @@ func GetTestMetadataForTgc(service, address, rawConfig string) resource.TestChec
 		return nil
 	}
 }
+
+// parseResources extracts all resources from a Terraform configuration string
+func parseResources(config string) []string {
+	// This regex matches resource blocks in Terraform configurations
+	resourceRegex := regexp.MustCompile(`resource\s+"([^"]+)"\s+"([^"]+)"`)
+	matches := resourceRegex.FindAllStringSubmatch(config, -1)
+
+	var resources []string
+	for _, match := range matches {
+		if len(match) >= 3 {
+			// Combine resource type and name to form the address
+			resources = append(resources, fmt.Sprintf("%s.%s", match[1], match[2]))
+		}
+	}
+
+	return resources
+}
+
+// getServicePackage determines the service package for a resource type
+func getServicePackage(resourceType string) string {
+	var ServicePackages = map[string]string{
+		"google_compute_":   "compute",
+		"google_storage_":   "storage",
+		"google_sql_":       "sql",
+		"google_container_": "container",
+		"google_bigquery_":  "bigquery",
+		"google_project":    "resourcemanager",
+		"google_cloud_run_": "cloudrun",
+	}
+
+	// Check for exact matches first
+	if service, ok := ServicePackages[resourceType]; ok {
+		return service
+	}
+
+	// Check for prefix matches
+	for prefix, service := range ServicePackages {
+		if strings.HasPrefix(resourceType, prefix) {
+			return service
+		}
+	}
+
+	// Default to "unknown" if no match found
+	return "unknown"
+}
+
+// extendWithTGCData adds TGC metadata check functions to each TestStep
+func extendWithTGCData(c resource.TestCase) resource.TestCase {
+	var updatedSteps []resource.TestStep
+
+	for _, step := range c.Steps {
+		if step.Config != "" {
+			// Parse resources from the config
+			resources := parseResources(step.Config)
+
+			// Create TGC metadata checks for each resource
+			var tgcChecks []resource.TestCheckFunc
+			for _, res := range resources {
+				parts := strings.Split(res, ".")
+				if len(parts) >= 2 {
+					resourceType := parts[0]
+					service := getServicePackage(resourceType)
```

Comment:
For the unknown service, I think we don't need to add the check function.  The `unknown` asset name will be filtered out before calling the cai assets API eventually.

Reply:
I would like to add the metadata for all tests so I can parse them for separate tasks. Maybe we can sync up here on how to be expose this metadata for both our usecases.

Reply:
If `unknown` is needed for your use cases, we can keep it. I can filter out them before calling the API.